### PR TITLE
Prefer magic link to password under a feature flag

### PR DESCRIPTION
**Pull request by jaclync**
Iteration 1 for #7437 

- [x] ⚠️ Please also review the WPAuthenticator PR https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/662, which contains most of the changes ⚠️ 
- [x] ⚠️ Make sure the WPAutheticator pod is using the production version in the Podfile ⚠️ 

### Description

We're experimenting with emphasizing the magic link in place of the password because the magic link has a much higher sign-in rate. In Iteration 1, we're automatically requesting a magic link right after the user enters their WP.com email and showing a new "magic link requested" screen with an alternative CTA to login with password. The design is in #7437 at hHmVq03kKpvTDzV3uV4wqw-fi-14%3A2092.

In the WPAuthenticator repo, the code changes are under a new configuration `WordPressAuthenticatorConfiguration.isWPComMagicLinkPreferredToPassword` and the app passes the value of a new feature flag `loginMagicLinkEmphasis`.

#### Login with WP.com

- Launch the app
- Log out if needed, and skip onboarding if needed
- Tap "Continue With WordPress.com"
- Enter a valid non-A8C WP.com email and tap "Continue" --> after a little bit (requesting magic link), a new magic link requested screen should be shown with two CTAs at the bottom
- Tap on "Open Mail" to make sure it works as before (on a device, the default mail app should be shown)
- Tap on "Use password to sign in" --> the password screen should be shown
- Continue with the password flow or tap on the magic link in the email to log in to the app

#### Login with site address

- Launch the app
- Log out if needed, and skip onboarding if needed
- Tap "Enter Your Store Address"
- Enter a valid WP site address and tap "Continue"
- Enter a valid non-A8C WP.com email and tap "Continue" --> after a little bit (requesting magic link), a new magic link requested screen should be shown with two CTAs at the bottom
- Tap on "Open Mail" to make sure it works as before (on a device, the default mail app should be shown)
- Tap on "Use password to sign in" --> the password screen should be shown
- Continue with the password flow or tap on the magic link in the email to log in to the app

### Screenshots

light | dark
-- | --
 | 

tablet:

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Authentication/AuthenticationManager.swift [hunk 1]
```diff
@@ -112,7 +114,7 @@ class AuthenticationManager: Authentication {
                                                               textColor: .text,
                                                               textSubtleColor: .textSubtle,
                                                               textButtonColor: .accent,
-                                                              textButtonHighlightColor: .accent,
+                                                              textButtonHighlightColor: .accentDark,
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Authentication/AuthenticationManager.swift:117]: Great catch. ❤️

File: Experiments/Experiments/DefaultFeatureFlagService.swift [hunk 2]
```diff
@@ -55,6 +55,8 @@ public struct DefaultFeatureFlagService: FeatureFlagService {
             return buildConfig == .localDeveloper || buildConfig == .alpha
         case .loginPrologueOnboardingSurvey:
             return true
+        case .loginMagicLinkEmphasis:
+            return true
```
Comment (selanthiraiyan) [inline Experiments/Experiments/DefaultFeatureFlagService.swift:59]: I believe that this means that we plan to release this iteration immediately. 

Sorry if this is mentioned somewhere already. Aren't we planning to use A/B testing? 🤔

**Review by selanthiraiyan — APPROVED** [review]
Nice work. 🎉 

It tests well. 

I left a couple of questions just for my learning purpose. 

-------

In the `Login with site address` flow

After the following step,
> Enter a valid non-A8C WP.com email and tap "Continue" --> after a little bit (requesting magic link), a new magic link requested screen should be shown with two CTAs at the bottom

the below event is tracked,
```
Tracked unified_login_interaction, properties: [AnyHashable("click"): "request_magic_link", AnyHashable("step"): "start", AnyHashable("flow"): "login_site_address", AnyHashable("source"): "default"]
```
Would it be alright to track a `click` property even though the user didn't manually request magic link?

-------

❓ Just curious
Do you have a sequence of analytics events planned to measure the success of this iteration? (To compare with other iterations and etc)

File: Experiments/Experiments/DefaultFeatureFlagService.swift (hunk 2 above)
Comment (jaclync) [reply Experiments/Experiments/DefaultFeatureFlagService.swift:59]: We don't have a way to A/B test in iOS yet (context pdnsEh-wr-p2#comment-1032), so we're just releasing experiments to all users for now 😞

**Comment by jaclync** [comment]
@selanthiraiyan I responded to your questions in the WPAuthenticator PR https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/662#pullrequestreview-1069411127 🙂

**Pull request merged**